packages/runtime/test-runtime-utils/src/mockStorage.ts

```diff
@@ -38,6 +38,27 @@ export class MockStorage implements IChannelStorageService {
         }
     }
 
+    private static readStringCore(tree: ITree, paths: string[]): string {
```

Comment:
Similar feedback as in other places - please try to avoid duplicating code.
Ideally all these cases should switch to implementing readBlob (instead of read), and readString being a tiny wrapper around it.
If you absolutely want to keep existing read() functionality, then you can make readCore to return IBlob and all callers to the last mile of conversion (including this API calling into readCore).